client/src/App.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import NavBar from './components/NavBar.jsx';
+import LineChartComponent from './components/LineChart.jsx';
+import BarChartComponent from './components/BarChart.jsx';
+
+
+class App extends React.Component {
+	constructor(props){
+		super(props);
+		this.state = {
+			organizationCriteria: 'alphabetical'
+		};
+	}
+
+	handleChartSelection(type) {
+		this.setState({chartType: type})
+	}
+
+	render() {
+
+		return (
+			<div>
+				<h1> Music Preferences Data </h1>
+				<div class="appContainer">
+					<div class="navContainer">
+						<NavBar selectChart={this.handleChartSelection.bind(this)}/>
+					</div>
+					<div class="chartContainer">
+					  { this.state.chartType === 'Listens per week'
+						?	<LineChartComponent />
```

Comment:
Potential Issues Here: What if we were to change what the text is, or what if we add a chart type, or what if we want more chart types to use the line chart component? How can we update this to be more generic/reusable?

Reply:
I see your point. One way to address this might be to leverage client-side routing, like with the 'react-router' library. That would make it considerably easier to add more charts (and routes) as needed. Since each chart is itself a component, we would be able to create intermediary components which make use of as many specific charts as we see fit, and map them to their own URLs via the router. I'm happy to do a quick mock-up of what I'm envisioning if that would help.